mod/primitives/pkg/math/u256_test.go

```diff
@@ -263,3 +264,277 @@ func TestUnwrapBigSign(t *testing.T) {
 		})
 	}
 }
+
+func TestMustNewU256L(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []byte
+		expected []byte
+		panic    bool
+	}{
+		{
+			name:     "valid input",
+			input:    []byte{1, 2, 3, 4, 5},
+			expected: []byte{5, 4, 3, 2, 1},
+			panic:    false,
+		},
+		{
+			name:  "invalid input - too long",
+			input: make([]byte, 33),
+			panic: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.panic {
+				require.Panics(t, func() {
+					math.MustNewU256L(tt.input)
+				})
+			} else {
+				result := math.MustNewU256L(tt.input)
+				expected := new(huint256.Int).SetBytes(tt.expected)
+				require.Equal(t, expected, result.UnwrapU256(), "Test case %s", tt.name)
+			}
+		})
+	}
+}
+
+func TestNewU256LFromBigEndian(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    []byte
+		expected []byte
+		err      error
+	}{
+		{
+			name:     "valid input",
+			input:    []byte{1, 2, 3, 4, 5},
```

Comment:
shouldnt this be flipping the bytes?

Reply:
@itsdevbear i think for case BigEndian shouldn't flipping the bytes